admin/configure.php

```diff
@@ -150,7 +150,7 @@ function civicrm_setup_instance(string $adminPath, bool $civicrmUpgrade): \Civi\
     if (defined('CIVICRM_DSN')) {
       $civiDSNParts = parse_url(CIVICRM_DSN);
       $model->db['username'] = $civiDSNParts['user'];
-      $model->db['password'] = $civiDSNParts['pass'];
+      $model->db['password'] = urldecode($civiDSNParts['pass']);
```

Comment:
@seamuslee001 Ah that sounds plausible. The corresponding `SettingsUtil::createParams()` and `DbUtil::encodeDsn()` both enable `urlencode()` for the password part. But they do similar encoding for username and database-name. I haven't `r-run`... but wouldn't it be appropriate for all three?

Reply:
@totten thanks for this, I have added it for all 4 parts now and also added a call to stripslashes because the params that get used in the dsn seem to have had addslashes done https://github.com/civicrm/civicrm-core/blob/master/setup/src/Setup/SettingsUtil.php#L18

Reply:
Sharp eye @seamuslee001. In this case, I think that `SettingsUtil` calls `addslashes()` to do roughly the job of `var_export()`. (`var_export()` would fell more idiomatically correct to me, except that `civicrm.settings.php.template` defines its own quotes.)

I haven't re-read the Joomla stuff, but it depends on how it's getting the original values:

* If one said `file_get_contents('civicrm.settings.php')` and started splitting strings, then you would need to stip out the slashes.
* If one said `require 'civicrm.settings.php'` and used the evaluated output, then you would not need to remove slashes. (*The PHP interpreter already did that.*)

Reply:
I think its the latter one https://github.com/civicrm/civicrm-joomla/blob/master/admin/configure.php#L182 but I think the point of the addslashes is to make sure `'` and `\` are properly escaped in the DSN  so if we are just parsing the DSN though would we not need to remove those additional backslashes but maybe i'm missing something in how the PHP interpreter is working?

Reply:
Aaah, calling `DB::parseDSN()` definitely sounds easier.  👍